### ISSUE TITLE
RawKeyboardListener returning incorrect keys on Windows 

### ISSUE THREAD

**Issue by esDotDev** [flutter/flutter#72533]
Seeing some odd behavior on Windows with my keys Insert,Home,Delete,End,PageUp,PageDown. 

This issue may have been closed pre-maturely:
https://github.com/flutter/flutter/issues/37710

I've tested on Windows and Web, the issue appears to only affect Windows. Logical keys look good, but Physical is randomly mapped to the numpad.
```
flutter: RawKeyDownEvent#19868(logicalKey: LogicalKeyboardKey#70049(keyId: "0x100070049", keyLabel: "", debugName: "Insert"), physicalKey: PhysicalKeyboardKey#70062(usbHidUsage: "0x00070062", debugName: "Numpad 0"))
flutter: RawKeyUpEvent#15314(logicalKey: LogicalKeyboardKey#70049(keyId: "0x100070049", keyLabel: "", debugName: "Insert"), physicalKey: PhysicalKeyboardKey#70062(usbHidUsage: "0x00070062", debugName: "Numpad 0"))
flutter: RawKeyDownEvent#489c5(logicalKey: LogicalKeyboardKey#7004a(keyId: "0x10007004a", keyLabel: "", debugName: "Home"), physicalKey: PhysicalKeyboardKey#7005f(usbHidUsage: "0x0007005f", debugName: "Numpad 7"))
flutter: RawKeyUpEvent#5112f(logicalKey: LogicalKeyboardKey#7004a(keyId: "0x10007004a", keyLabel: "", debugName: "Home"), physicalKey: PhysicalKeyboardKey#7005f(usbHidUsage: "0x0007005f", debugName: "Numpad 7"))
flutter: RawKeyDownEvent#e1b43(logicalKey: LogicalKeyboardKey#7004c(keyId: "0x10007004c", keyLabel: "", debugName: "Delete"), physicalKey: PhysicalKeyboardKey#70063(usbHidUsage: "0x00070063", debugName: "Numpad Decimal"))
flutter: RawKeyUpEvent#5d937(logicalKey: LogicalKeyboardKey#7004c(keyId: "0x10007004c", keyLabel: "", debugName: "Delete"), physicalKey: PhysicalKeyboardKey#70063(usbHidUsage: "0x00070063", debugName: "Numpad Decimal"))
flutter: RawKeyDownEvent#472e5(logicalKey: LogicalKeyboardKey#7004d(keyId: "0x10007004d", keyLabel: "", debugName: "End"), physicalKey: PhysicalKeyboardKey#70059(usbHidUsage: "0x00070059", debugName: "Numpad 1"))
flutter: RawKeyUpEvent#57913(logicalKey: LogicalKeyboardKey#7004d(keyId: "0x10007004d", keyLabel: "", debugName: "End"), physicalKey: PhysicalKeyboardKey#70059(usbHidUsage: "0x00070059", debugName: "Numpad 1"))
flutter: RawKeyDownEvent#fc6ac(logicalKey: LogicalKeyboardKey#00021(keyId: "0x10700000021", keyLabel: "", debugName: "Unknown Windows key code 33"), physicalKey: PhysicalKeyboardKey#70061(usbHidUsage: "0x00070061", debugName: "Numpad 9"))
flutter: RawKeyUpEvent#6a293(logicalKey: LogicalKeyboardKey#00021(keyId: "0x10700000021", keyLabel: "", debugName: "Unknown Windows key code 33"), physicalKey: PhysicalKeyboardKey#70061(usbHidUsage: "0x00070061", debugName: "Numpad 9"))
flutter: RawKeyDownEvent#0321c(logicalKey: LogicalKeyboardKey#00022(keyId: "0x10700000022", keyLabel: "", debugName: "Unknown Windows key code 34"), physicalKey: PhysicalKeyboardKey#7005b(usbHidUsage: "0x0007005b", debugName: "Numpad 3"))
flutter: RawKeyUpEvent#d9b02(logicalKey: LogicalKeyboardKey#00022(keyId: "0x10700000022", keyLabel: "", debugName: "Unknown Windows key code 34"), physicalKey: PhysicalKeyboardKey#7005b(usbHidUsage: "0x0007005b", debugName: "Numpad 3"))
```

Is it expected that there is no physical delete/insert/pageUp etc? Keyboard is a full desktop keyboard from Corsair:
![image](https://user-images.githubusercontent.com/736973/102527884-067d2b00-405b-11eb-9160-ed401bbc1a1e.png)

I have not tested this on MacOS or Linux. 

```
[√] Flutter (Channel master, 1.26.0-2.0.pre.69, on Microsoft Windows [Version 10.0.19041.630], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.6)
[√] Android Studio (version 4.0)
[√] VS Code (version 1.52.1)
[√] Connected device (3 available)

• No issues found!
```

**Comment by esDotDev** [comment]
Some other interesting results:
PrintScreen gives me:
`flutter: RawKeyUpEvent#f89b9(logicalKey: LogicalKeyboardKey#0002c(keyId: "0x1070000002c", keyLabel: "", debugName: "Unknown Windows key code 44"), physicalKey: PhysicalKeyboardKey#70055(usbHidUsage: "0x00070055", debugName: "Numpad Multiply"))`
NumLock, gives me: 
`RawKeyUpEvent#642e2(logicalKey: LogicalKeyboardKey#00090(keyId: "0x10700000090", keyLabel: "", debugName: "Unknown Windows key code 144"), physicalKey: PhysicalKeyboardKey#70048(usbHidUsage: "0x00070048", debugName: "Pause"))`
ContextMenu key gives me this:
`RawKeyDownEvent#8464b(logicalKey: LogicalKeyboardKey#0005d(keyId: "0x1070000005d", keyLabel: "", debugName: "Unknown Windows key code 93"), physicalKey: PhysicalKeyboardKey#00000(usbHidUsage: "0x00000000", debugName: "None"))`
ControlRight gives me ControlLeft (As does Alt-Right): 
`RawKeyUpEvent#426d3(logicalKey: LogicalKeyboardKey#700e0(keyId: "0x1000700e0", keyLabel: "", debugName: "Control Left"), physicalKey: PhysicalKeyboardKey#700e0(usbHidUsage: "0x000700e0", debugName: "Control Left"))`

As far as I can tell, all others are correct.

**Comment by TahaTesser** [comment]
Hi @esDotDev 
Thanks for filing the issue, I can reproduce it on Windows

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  FocusNode _focusNode = new FocusNode();

  @override
  void dispose() {
    _focusNode.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: RawKeyboardListener(
          focusNode: _focusNode,
          onKey: (RawKeyEvent event) {
            print(event);
          },
          child: TextField(),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

``` 

</details>

<details> 
<summary>multiple flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel dev, 1.26.0-1.0.pre, on Microsoft Windows [Version 10.0.19042.685], locale en-US)
    • Flutter version 1.26.0-1.0.pre at C:\Code\flutter_dev
    • Framework revision 63062a6443 (5 days ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30804.86
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.685]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 87.0.664.60

• No issues found!
```

```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.84, on Microsoft Windows [Version 10.0.19042.685], locale en-US)
    • Flutter version 1.26.0-2.0.pre.84 at C:\Code\flutter_master
    • Framework revision cda1fae6b6 (3 hours ago), 2020-12-18 04:04:02 -0500
    • Engine revision 1be6f414e7
    • Dart version 2.12.0 (build 2.12.0-166.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30804.86
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.685]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 87.0.664.60

• No issues found!
```

</details>

**Comment by dkwingsmt** [comment]
This will very likely be fixed by https://github.com/flutter/flutter/issues/44918.

**Comment by dkwingsmt** [comment]
Progress:
- The following keys are working correctly on the current master (41ff30c87111e393545c16df937492705e890472): Home, End, Delete, PageUp, PageDown, NumLock, ContextMenu
- The following key does not seem to be possible, since the messages are not dispatched to Flutter: Insert
- The following keys should be fixed once https://github.com/flutter/engine/pull/27921 is merged and rolled in: ControlRight, AltRight
- The following keys still have problems: PrintScreen

**Comment by gspencergoog** [comment]
Another data point: When I run a Windows app, and press "Insert" using the master channel, it crashes the app.

When it crashes, in the (Dart) exception, it gives the key event as:

```
RawKeyDownEvent#a2ba8(logicalKey: LogicalKeyboardKey#000ff(keyId: "0x16000000ff", keyLabel: "",
debugName: "Key with ID 0x016000000ff"), physicalKey: PhysicalKeyboardKey#70053(usbHidUsage:
"0x00070053", debugName: "Num Lock")) and its data: RawKeyEventDataWindows#f7e83(keyCode: 255,
scanCode: 57413, characterCodePoint: 0, modifiers: 0)
```

Which is weird to me, since it isn't the NumLock key.

255 seems like it could be some kind of "error" keycode.  The keycode for Insert should be 0x2d

It's probably crashing because it looks at "NumLock" and thinks it needs to synchronize modifiers, but no modifier flags are set (because it's not a modifier).

**Comment by dkwingsmt** [comment]
@gspencergoog 0xff is set for all "unknown keys" (https://stackoverflow.com/a/48132748). We probably should ignore such keys, or at least mint them with scancode.

Nevertheless, I can't reproduce this issue. If it does, it should be resolved if we lift the limitation of https://github.com/flutter/flutter/pull/93897.

**Comment by gspencergoog** [comment]
I am still able to reproduce this with the latest master (b865e23aeb).

**Comment by dkwingsmt** [comment]
@gspencergoog Can you show your app? I tried to build one with `RawKeyListener` and didn't hit any crashes.

**Comment by gspencergoog** [comment]
It might by a separate issue.  I'll file a new issue with repro steps, and we can track it there.

**Comment by gspencergoog** [comment]
Separate issue is https://github.com/flutter/flutter/issues/94502

**Comment by dkwingsmt** [comment]
As of latest (43c81ff) this problem is no longer reproducible.
